lib/refile/backend/file_system.rb

```diff
@@ -92,6 +92,22 @@ def initialize(directory, max_size: nil, hasher: Refile::RandomHasher.new)
         ::File.exist?(path(id))
       end
 
+      # Return eTag of the file
+      #
+      # @param [Sring] id           The id of the file
+      # @return [String]
+      verify_id def etag(id)
+        Digest::MD5.file(path(id)).hexdigest
```

Comment:
I think we should at least use SHA256. There is probably no way to exploit this, but just to be sure. MD5 is known to be broken. I can't imagine that the difference in speed is really going to make a difference.